modules/javafx.graphics/src/main/java/javafx/stage/Window.java

```diff
@@ -1547,10 +1547,12 @@ public void apply() {
                 float newRX = (float) renderScaleX;
                 float newRY = (float) renderScaleY;
                 reset();
-                peer.setBounds(newX, newY, xSet, ySet,
-                                    newWW, newWH, newCW, newCH,
-                                    newXG, newYG,
-                                    newRX, newRY);
+                if (peer != null) {
+                    peer.setBounds(newX, newY, xSet, ySet,
+                            newWW, newWH, newCW, newCH,
+                            newXG, newYG,
+                            newRX, newRY);
+                }
```

Comment:
The set of local variables will not be needed when peer is null.
Please consider change as below,

public void apply() {
            if (dirty) {
                if (peer != null) {
                    // Snapshot values and then reset() before we call down
                    // as we may end up with recursive calls back up with
                    // new values that must be recorded as dirty.
                    boolean xSet = !Double.isNaN(x);
                    float newX = xSet ? (float) x : 0f;
                    boolean ySet = !Double.isNaN(y);
                    float newY = ySet ? (float) y : 0f;
                    float newWW = (float) windowWidth;
                    float newWH = (float) windowHeight;
                    float newCW = (float) clientWidth;
                    float newCH = (float) clientHeight;
                    float newXG = xGravity;
                    float newYG = yGravity;
                    float newRX = (float) renderScaleX;
                    float newRY = (float) renderScaleY;
                    reset();
                    peer.setBounds(newX, newY, xSet, ySet,
                                        newWW, newWH, newCW, newCH,
                                        newXG, newYG,
                                        newRX, newRY);
                } else {
                    reset();
                }
            }
        }

Reply:
A second alternative would be to return early from the method if peer is null, leaving the rest of the method unchanged.

```
    public void apply() {
        if (dirty) {
            if (peer == null) {
                reset();
                return;
            }
            ...
        }
```

I don't have a strong opinion. Any of the three choices (the original fix, Ambarish's proposed change, or my variation on that) are fine with me.

Reply:
> A second alternative would be to return early from the method if peer is null, leaving the rest of the method unchanged.
> 
> ```
>     public void apply() {
>         if (dirty) {
>             if (peer == null) {
>                 reset();
>                 return;
>             }
>             ...
>         }
> ```
> I don't have a strong opinion. Any of the three choices (the original fix, Ambarish's proposed change, or my variation on that) are fine with me.

I find the proposal from @kevinrushforth clearer.